tests/foreman/api/test_errata.py

```diff
@@ -183,31 +184,53 @@ def test_positive_install_multiple_in_host(
 
     :CaseLevel: System
     """
+    ak = target_sat.api.ActivationKey(
+        organization=function_org,
+        environment=function_lce,
+    ).create()
+    cv = target_sat.api.ContentView(organization=function_org).create()
+    # Associate custom repos with org, cv, lce, ak:
+    target_sat.cli_factory.setup_org_for_a_custom_repo(
+        {
+            'url': settings.repos.yum_9.url,
+            'organization-id': function_org.id,
+            'content-view-id': cv.id,
+            'lifecycle-environment-id': function_lce.id,
+            'activationkey-id': ak.id,
+        }
+    )
+    # Install katello-ca, register content-host, enable all repos:
     rhel_contenthost.install_katello_ca(target_sat)
-    rhel_contenthost.register_contenthost(module_org.label, activation_key.name)
+    rhel_contenthost.register_contenthost(function_org.name, ak.name, function_lce.name)
     assert rhel_contenthost.subscribed
+    # Installing outdated custom packages:
     for package in constants.FAKE_9_YUM_OUTDATED_PACKAGES:
+        rhel_contenthost.run(f'yum remove -y {str(package.split("-", 1)[0])}')
         assert rhel_contenthost.run(f'yum install -y {package}').status == 0
-    applicable_errata_count = rhel_contenthost.applicable_errata_count
-    assert applicable_errata_count > 1
+        assert rhel_contenthost.run(f'rpm -q {package}').status == 0
     rhel_contenthost.add_rex_key(satellite=target_sat)
-    for errata in settings.repos.yum_9.errata[1:4]:
+    rhel_contenthost.run(r'subscription-manager repos --enable \*')
+    # Each errata will be installed sequentially,
+    # after each install, applicable-errata-count should drop by one.
+    for errata in constants.FAKE_9_YUM_SECURITY_ERRATUM:
+        pre_errata_count = copy(rhel_contenthost.applicable_errata_count)
+        assert pre_errata_count >= 1
         task_id = target_sat.api.JobInvocation().run(
             data={
                 'feature': 'katello_errata_install',
                 'inputs': {'errata': str(errata)},
                 'targeting_type': 'static_query',
                 'search_query': f'name = {rhel_contenthost.hostname}',
-                'organization_id': module_org.id,
+                'organization_id': function_org.id,
             },
         )['id']
         target_sat.wait_for_tasks(
             search_query=(f'label = Actions::RemoteExecution::RunHostsJob and id = {task_id}'),
             search_rate=20,
             max_tries=15,
         )
-        applicable_errata_count -= 1
-        assert rhel_contenthost.applicable_errata_count == applicable_errata_count
+        sleep(10)
```

Comment:
What's the reasoning for this sleep? Can we do a wait_for on the assert instead? Sleep shouldn't be necessary here.

Reply:
A wait_for on the assert was consistently failing prt, and inconsistent fail locally. 
I had tried something like this :
`wait_for(lambda: rhel_ch.applicable_errata_count == app_err_count - 1, timeout=600, delay=10)`
But would occasionally fail to complete lambda in time?

Reply:
The sleep method was the most consistent way for prt to pass. Two methods doing similar tasks in api:errata, `_fetch_available_errata` and `_validate_errata_counts` are using `sleep()` before getting counts

Reply:
Since it looks like this task is installing an errata and then immediately checking the new errata status, I think the sleep makes sense since it could take some time for that whole orchestration to finish.

The host errata install task completes, so the info should also be reported back, but the applicable errata calculation is asynchronous and may not be complete.

The best solution would be to poll on the errata applicability calculation, but we may not have the data to determine exactly which task to poll.

Reply:
And I mean sleep or some sort of delay, I'm not sure what the `wait_for` is that @sambible mentioned.

Reply:
I'm fine in this case then. Sleep can be a decent tool in certain cases, and it sounds like this one is in that situation.

Reply:
Thanks for the insight @ianballou! `wait_for` is a method we use that will fail if a condition `lambda:` could not be met within a specified timeout (seconds), delay is the time between each check. 

Lots of tests in robottelo use this and I did so recently in `report_templates:installable_errata` test here:
https://github.com/SatelliteQE/robottelo/blob/0ddb4b712f659b78791e5a6437dca4bf140e8ce6/tests/foreman/api/test_reporttemplates.py#L809